src/Sylius/Bundle/CoreBundle/Controller/ProductController.php

```diff
@@ -114,6 +115,36 @@ public function indexByTaxonIdAction(Request $request, $id)
         return $this->renderResults($taxon, $paginator, 'productIndex.html', $request->get('page', 1));
     }
 
+    /**
+     * List products categorized under given archetype (fetch by its code or id).
+     *
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws NotFoundHttpException
+     */
+    public function indexByArchetypeAction(Request $request)
+    {
+        $archetypeIdOrCode = $request->attributes->get('archetype');
+        if (filter_var($archetypeIdOrCode, FILTER_VALIDATE_INT)) {
+            $archetype = $this->get('sylius.repository.product_archetype')->find($archetypeIdOrCode);
+        } else {
+            $archetype = $this->get('sylius.repository.product_archetype')->findOneByCode($archetypeIdOrCode);
+        }
```

Comment:
@Zales0123 Have a look on @aramalipoor second comment of this PR, I've already done this but he said that this would be better in one action. Please discuss this and let me know how finally should I do this (You wouldn't find it in commits because I have to reset git soft because of mistake in pushing commits).